src/__tests__/spectral.test.ts

```diff
@@ -51,6 +51,129 @@ describe('spectral', () => {
     expect(expectedCustomRuleSet).toEqual(givenCustomRuleSet);
   });
 
+  // Assures: https://stoplightio.atlassian.net/browse/SL-789
+  test('setting rules with opts should update/append the current ruleset', () => {
+    const ruleset = {
+      rules: {
+        format: {
+          rule1: {
+            type: RuleType.STYLE,
+            function: RuleFunction.TRUTHY,
+            path: '$',
+            enabled: true,
+            summary: '',
+            input: {
+              properties: 'something',
+            },
+          },
+        },
+      },
+    };
+    // deep copy
+    const s = new Spectral({ rulesets: [ruleset] });
+
+    s.setRules([
+      {
+        rules: {
+          differentFormat: {
+            rule2: {
+              type: RuleType.STYLE,
+              function: RuleFunction.TRUTHY,
+              path: '$',
+              enabled: true,
+              summary: '',
+              input: {
+                properties: 'a different rule',
+              },
+            },
+          },
+        },
+      },
+    ]);
+
+    expect(s.getRules()).toHaveLength(2);
+
+    s.setRules(
+      [
+        {
+          rules: {
+            format: {
+              rule1: false,
+            },
+          },
+        },
+      ],
+      { includeCurrent: true }
```

Comment:
That'll do the trick!

What you did here reminds me something widely known as the [Flag Argument](https://martinfowler.com/bliki/FlagArgument.html). 

I'm curious what you think about another approach to solving that type of logic. 

One idea, that I'm sure you must have thought about, would be to split the implementation and expose a second function called: `updateRules`. Such method would merge with existing ruleset whereas the `setRules` would override it. 

One `if` statement less :)

With regards,
Chris The If Statement Slayer

Reply:
I considered having two separate functions as well, but the internal logic in them felt repetitious. It's actually exactly the same except one passes includeCurrent to _parseRulesets, while one does not

Reply:
Separate function could still be a good idea for a clear understanding of what the functions do based on naming convention. And to have a clearer differentiation of responsibility. @chris-miaskowski @marbemac thoughts?

Reply:
@casserni how about exposing two different public method and using a shared private function to implement them?